### ISSUE TITLE
Ignore specific expressions in `LabeledExpression`

### ISSUE THREAD

**Issue by chezwicker** [detekt/detekt#6118]
## Expected Behavior

Allow specifying e.g. an array `ignoreExpressions` for `LabeledExpression`.

## Current Behavior

It's possible to ignore labels, but not specific expressions themselves.

## Context

We'd like to be able to partially allow `LabeledExpression` for only `this@`, but not in other cases.

**Comment by cortinico** [comment]
Can you provide an example?

**Comment by chezwicker** [comment]
You mean for when I'd want to use `this@`? Sure (adapted from real life, but close match):

```kotlin
fun State.assignShippingNumbers(itemId: ItemId): State {
    val shippingDate = orders.getValue(orderId).getShippingDate()
    return updateItems(itemId) {
        assignShippingNumber(this@assignShippingNumbers, shippingDate)
    }
}
```

At the same time, we don't want to use labels to jump out of loops.

**Comment by chezwicker** [comment]
Can this bot be configured to only mark an issue as stale if the last comment was by a contributor? I'm not quite sure how I could add to it in a useful way.

**Comment by BraisGabin** [comment]
I would say that this boy also help us to drag attention over this issues that weren't answered. But we could open an issue or conversation to talk about the bot.

About the issue: I'm not sure if it is worth the effort to implement and maintain this. The idea of this rule is that the labeled expressions are more difficult to follow. And I think that your example is not an exception. If there is not other way, that's a good reason for a suppress. But I see different options:
- Don't use the same name for the extension and the method. 
- Make the extension function a top level extension function. This way you don't even have the option to use `@this`.

So, to me this rule is doing what is meant to do so I wouldn't add it. Anyway you can always copy the rule and modify it to your needs as a third party rule.

**Comment by BraisGabin** [comment]
Closing this for the reasons described above.
